src/dev.sh

```diff
@@ -304,13 +304,21 @@ function package ()
         tar_name="${agent_pkg_name}.tar.gz"
         echo "Creating $tar_name in ${LAYOUT_DIR}"
         tar -czf "${tar_name}" -C ${LAYOUT_DIR} .
+        echo "Calculate SHA256 checksum for $tar_name"
+        if [[ ("$PLATFORM" == "linux") ]]; then
+            sha256sum "${tar_name}" | cut -b 1-64 > ./Checksum_SHA256.txt
```

Comment:
@ericsciple The approach "They can download to a temp location and then move to the cache location if the download is successful" generally might work but would add extra burden on the users and is not 100% fail-proof either.

Regarding the latter: just now I was downloading `vsts-agent-rhel.7.2-x64-2.104.0.tar.gz` from github. Everything was reported to be successful. But unfortunately github had hiccups just at that moment and instead served and HTML slate page (service not available with angry unicorn). So the end result was not a proper `tar.gz`. This of course led to the script bombing out later. 

I guess I could use `file` to check that we actually have a `tar.gz` or add some other kind of logic but this makes things more complicated than simply checking against an expected checksum and retrying if that does not match.